server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/InternalGeoHashGrid.java

```diff
@@ -63,26 +65,29 @@
          * Read from a stream.
          */
         private Bucket(StreamInput in) throws IOException {
+            type = GeoHashType.readFromStream(in);
             geohashAsLong = in.readLong();
             docCount = in.readVLong();
             aggregations = InternalAggregations.readAggregations(in);
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
+            type.writeTo(out);
```

Comment:
We need to check the version here

Reply:
The version is checked inside the type itself -- https://github.com/elastic/elasticsearch/pull/30320/files#diff-75ad5ae5ac78babf26799166a287cfe1R51

Reply:
See above